src/main/java/codeu/controller/LoginServlet.java

```diff
@@ -59,27 +59,41 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
   }
 
   /**
-   * This function fires when a user submits the login form. It gets the username from the submitted
-   * form data, and then adds it to the session so we know the user is logged in.
+   * This function fires when a user submits the login form. It gets the username and password from the submitted
+   * form data, checks that they're valid, and then either adds the user to the session so we know the user is logged in
+   * or shows an error to the user.
    */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
     String username = request.getParameter("username");
+    String password = request.getParameter("password");
+
+    if (userStore.isUserRegistered(username)) {
+        User user = userStore.getUser(username);
+    
+    if(BCrypt.checkpw(password,user.getPassword())) {
```

Comment:
indent of your if, else logic here makes the code hard to parse.  Review style guide and apply to the code to make it more readable.